File: Models/Reclamation.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace MiniProjet.Models
{
    public class Reclamation
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public DateTime DateReclamation { get; set; }
        public int idArticleReclamation { get; set; }

        [ForeignKey("idArticleReclamation")]
        public Article? article {  get; set; }   
        public int EtatId { get; set; }

        
        public int ClientId { get; set; }

        [ForeignKey("ClientId")]
        public Client? Client { get; set; }
        public int? InterventionId { get; set; }

        

    }

}

File: src/app/models/reclamation.ts
export interface Reclamation {
    id: number;
    description: string;
    dateReclamation: string;
    idArticleReclamation: number;
    article?: {
      id: number;
      libelle: string;
      estSousGarantie: boolean;
      prix: number;
    };
    etatId: number;
    clientId: number;
    client?: {
      id: number;
      firstName: string;
      lastName: string;
      uSerName: string;
      password: string;
    };
    interventionId?: number;
  }
  